libraries/AP_WindVane/AP_WindVane.cpp

```diff
@@ -0,0 +1,320 @@
+    // @User: Standard
+    AP_GROUPINFO("ANA_PIN", 3, AP_WindVane, _analog_pin_no, WINDVANE_DEFAULT_PIN),
+
+    // @Param: ANA_V_MIN
+    // @DisplayName: Analog minumum voltage
+    // @Description: Minimum analalog voltage read by windvane
+    // @Units: V
+    // @Increment: 0.01
+    // @Range: 0 5.0
+    // @User: Standard
+    AP_GROUPINFO("ANA_V_MIN", 4, AP_WindVane, _analog_volt_min, 0.0f),
+
+    // @Param: ANA_V_MAX
+    // @DisplayName: Analog maximum voltage
+    // @Description: Minimum analalog voltage read by windvane
+    // @Units: V
+    // @Increment: 0.01
+    // @Range: 0 5.0
+    // @User: Standard
+    AP_GROUPINFO("ANA_V_MAX", 5, AP_WindVane, _analog_volt_max, 3.3f),
+
+    // @Param: ANA_OF_HD
+    // @DisplayName: Analog headwind offset
+    // @Description: Angle offset when windvane is indicating a headwind, ie 0 degress relative to vehicle
+    // @Units: deg
+    // @Increment: 1
+    // @Range: 0 360
+    // @User: Standard
+    AP_GROUPINFO("ANA_OF_HD", 6, AP_WindVane, _analog_head_bearing_offset, 0.0f),
+
+    // @Param: VANE_FILT
+    // @DisplayName: Wind vane low pass filter frequency
+    // @Description: Wind vane low pass filter frequency, a value of -1 disables filter
+    // @Units: Hz
+    // @User: Standard
+    AP_GROUPINFO("VANE_FILT", 7, AP_WindVane, _vane_filt_hz, 0.5f),
+
+    // @Param: CAL
+    // @DisplayName: set to one to enter calibration on reboot
+    // @Description: set to one to enter calibration on reboot
+    // @Values: 0:None, 1:Calibrate
+    // @User: Standard
+    AP_GROUPINFO("CAL", 8, AP_WindVane, _calibration, 0),
+
+    // @Param: ANA_DZ
+    // @DisplayName: Analog potentiometer dead zone
+    // @Description: Analog potentiometer mechanical dead zone
+    // @Units: deg
+    // @Increment: 1
+    // @Range: 0 360
+    // @User: Standard
+    AP_GROUPINFO("ANA_DZ", 9, AP_WindVane, _analog_deadzone, 0),
+
+    AP_GROUPEND
+};
+
+// create a global instances of low pass filter
+LowPassFilterFloat low_pass_filter_wind_sin = LowPassFilterFloat(2.0f);
+LowPassFilterFloat low_pass_filter_wind_cos = LowPassFilterFloat(2.0f);
+LowPassFilterFloat low_pass_filter_wind_speed = LowPassFilterFloat(2.0f);
+
+// Public
+// ------
+
+// constructor
+AP_WindVane::AP_WindVane()
+{
+    AP_Param::setup_object_defaults(this, var_info);
+    if (_s_instance) {
+        AP_HAL::panic("Too many Wind Vane sensors");
+    }
+    _s_instance = this;
+}
+
+// destructor
+AP_WindVane::~AP_WindVane(void)
+{
+}
+
+/*
+ * Get the AP_WindVane singleton
+ */
+AP_WindVane *AP_WindVane::get_instance()
+{
+    return _s_instance;
+}
+
+// return true if wind vane is enabled
+bool AP_WindVane::enabled() const
+{
+    return (_type != WINDVANE_NONE);
+}
+
+// Initialize the Wind Vane object and prepare it for use
+void AP_WindVane::init()
+{
+    // a pin for reading the Wind Vane voltage
+    windvane_analog_source = hal.analogin->channel(ANALOG_INPUT_NONE);
+}
+
+// Update wind vane, called at 10hz
```

Comment:
Probably worth rewording this to "expected to be called at a minimum of 10 hz". In general we are trying to avoid having things hard coded against a tick rate.